torch/_functorch/partitioners.py

```diff
@@ -807,11 +807,29 @@ def solve_min_cut(
         print("Ops banned from rematerialization: ", ops_ignored)
         print()
 
+    def can_fuse_into_auto_functionalized(a, b):
+        if b.target != torch.ops.higher_order.auto_functionalized:
+            return False
+        mutable_op = b.args[0]
+        mutable_arg_names = (
+            torch._higher_order_ops.auto_functionalize.get_mutable_arg_names(mutable_op)
+        )
+        for name in mutable_arg_names:
+            arg = b.kwargs[name]
+            if a is arg:
+                return True
+            if isinstance(arg, list):
```

Comment:
should this be a tree_map or something?

Reply:
We only support TensorList args as inputs to operators, so the list is fine